src/objectExplorer/nodes/connectionNode.ts

```diff
@@ -59,6 +62,82 @@ export class ConnectionNode extends TreeNodeInfo {
         } else {
             this.iconPath = ObjectExplorerUtils.iconPath(ICON_SERVER_DISCONNECTED);
         }
+
+        // Tooltip logic: show all non-default properties except those in the label (database, user, server)
+        const connectionTooltip = this.getConnectionTooltip(connectionProfile);
+        if (connectionTooltip) {
+            this.tooltip = connectionTooltip;
+        }
+    }
+
+    private getConnectionTooltip(connectionProfile: IConnectionProfile): string | undefined {
+        // Properties to exclude
+        const exclude = [
+            "database",
+            "user",
+            "server",
+            "profileName",
+            "id",
+            "groupId",
+            "profileSource",
+            "savePassword",
+            "emptyPasswordInput",
+            "azureAuthType",
+            "password",
+        ];
+        // Default values for comparison
+        const connectionNodeDefaults: Partial<IConnectionProfile> = getDefaultConnection();
+
+        const defaultValues = {
+            encrypt: (connectionNodeDefaults as any).encrypt,
+            trustServerCertificate: (connectionNodeDefaults as any).trustServerCertificate,
+            persistSecurityInfo: (connectionNodeDefaults as any).persistSecurityInfo,
+            azureAuthType: (connectionNodeDefaults as any).azureAuthType,
+            multipleActiveResultSets: (connectionNodeDefaults as any).multipleActiveResultSets,
+            connectTimeout: (connectionNodeDefaults as any).connectTimeout,
+            commandTimeout: (connectionNodeDefaults as any).commandTimeout,
+            applicationName: (connectionNodeDefaults as any).applicationName,
+            savePassword: (connectionNodeDefaults as any).savePassword,
+            emptyPasswordInput: (connectionNodeDefaults as any).emptyPasswordInput,
+            profileSource: (connectionNodeDefaults as any).profileSource,
+            authenticationType: (connectionNodeDefaults as any).authenticationType,
+            applicationIntent: (connectionNodeDefaults as any).applicationIntent,
+        };
+
+        let props: string[] = [];
+
+        //handle auth types properly, if auth type is integrated or azureMfa don't show user in tooltip
+        if (
+            connectionProfile.authenticationType === Constants.integratedauth ||
+            connectionProfile.authenticationType === Constants.azureMfa
+        ) {
+            exclude.push("user");
+        }
+
+        // if connection has a profile name, don't exclude server, user and database from tooltip
+        if (connectionProfile.profileName) {
```

Comment:
Lets include these details even if profile name isn't present. A lot of times database and user gets truncated from the autogenerated name. The only way to see everything will be to resize OE (which isn't the best experience)